File: rust/src/bin/p0097.rs
// Project Euler: Problem 97

euler::run_solver!(97);

fn solve() -> String {
    compute(10)
}

fn compute(n_digits: u32) -> String {
    use euler::math;

    let divisor = 10_i64.pow(n_digits);
    format!(
        "{:0width$}",
        (28433 * math::powmod(2, 7830457, divisor) + 1) % divisor,
        width = n_digits as usize
    )
}

#[cfg(test)]
mod tests {
    use super::compute;

    #[test]
    fn p0097() {
        assert_eq!(compute(10), "8739992577");
    }
}

File: ocaml/src/p0097.ml
(* Project Euler: Problem 97 *)

open Core

let compute n_digits =
  let modulus = Int.pow 10 n_digits in
  Printf.sprintf
    "%0*d"
    n_digits
    (((28433 * Euler.Math.powmod 2 7830457 modulus) + 1) mod modulus)
;;

let solve () = compute 10

(* Test *)

let%test_unit "10" = [%test_eq: string] (compute 10) "8739992577"
